src/backend/aspen/database/models/sample.py

```diff
@@ -76,10 +76,12 @@ def create_public_ids(
         next_id = 0
     next_id += 1
     ids: List[str] = []
+    # some group prefixes already have a dash, but if they don't add a dash to the end
```

Comment:
Will there be any issues elsewhere with an ID-->group parser or something? I'm thinking if we _always_ assume that the group prefix would be its literal prefix (including dash), and that would cause problems with something where the "true" prefix is, say `abc-`, but we parse an ID that is `hCoV-19/usa/abc-SomeNextID/2021`, something else would barf because it would read the group's prefix as `abc` instead of its true value of `abc-`? It's definitely an edge case, but I just don't know if the code base ever reads the ID the other direction to extract info.

Reply:
Similarly, would we ever have to worry about two groups where one is `abc` and the other is `abc-`? If that happens, then we could get identical IDs for different groups. Don't know if that's a problem, or if we're sure we won't have that kind of overlap in group prefixes.

Reply:
i don't think there would ever be a case where there would be 2 groups (one with a dash and one without a dash), aspen admins are supplying the group prefixes directly in the DB, so this would not be an issue (but maybe something to consider if groups are allowed to decide on their own prefixes in the future).